models/users.js

```diff
@@ -397,37 +407,150 @@ const addForVerification = async (userId, discordId, profileImageUrl, discordIma
  * @return {Promise<{message: string}|{message: string}>}
  * @throws {Error} - If error occurs while verifying user's image
  */
-const markAsVerified = async (userId, imageType) => {
+const changePhotoVerificationStatus = async (userId, imageType, status) => {
   try {
-    const verificationUserDataSnapshot = await photoVerificationModel.where("userId", "==", userId).get();
+    const verificationUserDataSnapshot = await photoVerificationModel
+      .where("userId", "==", userId)
+      .where("status", "==", photoVerificationRequestStatus.PENDING)
+      .get();
+
     // THROWS ERROR IF NO DOCUMENT FOUND
     if (verificationUserDataSnapshot.empty) {
       throw new Error("No verification document record data for user was found");
     }
-    // VERIFIES BASED ON THE TYPE OF IMAGE
-    const imageVerificationType = imageType === "discord" ? "discord.approved" : "profile.approved";
+
     const documentRef = verificationUserDataSnapshot.docs[0].ref;
-    await documentRef.update({ [imageVerificationType]: true });
-    return { message: "User image data verified successfully" };
+
+    // if status is rejected then remove the verification entry
+    if (status === photoVerificationRequestStatus.REJECTED) {
+      await documentRef.update({ status });
+      return "User photo verification request rejected successfully";
+    }
+
+    // VERIFIES BASED ON THE TYPE OF IMAGE
+    if (imageType === "both") {
+      await documentRef.update({ "discord.approved": true, "profile.approved": true });
+    } else {
+      const imageVerificationType = imageType === "discord" ? "discord.approved" : "profile.approved";
+      await documentRef.update({ [imageVerificationType]: true });
+    }
+
+    // IF BOTH IMAGES ARE VERIFIED THEN REMOVE THE ENTRY
+    const photoVerificationObject = (await documentRef.get()).data();
+    if (photoVerificationObject.discord.approved && photoVerificationObject.profile.approved) {
+      await documentRef.update({ status: photoVerificationRequestStatus.APPROVED });
+      await updateUserPicture(
+        { url: photoVerificationObject.profile.url, publicId: photoVerificationObject.profile.publicId },
+        userId
+      );
```

Comment:
Do both the call needs to happen in a series?

Reply:
yes, first the `photo-verification-object` needs to be updated, then the user picture needs to be updated too.